app/batid/services/data_gouv_publication.py

```diff
@@ -11,22 +11,32 @@
 from django.db import connection
 
 
-def publish():
+def publish(areas_list):
     # Publish the RNB on data.gouv.fr
     directory_name = create_directory()
+    print(str(len(areas_list)) + " area(s) to process...")
 
     try:
-        create_rnb_csv_files(directory_name)
-        (archive_path, archive_size, archive_sha1) = create_archive(directory_name)
-        public_url = upload_to_s3(archive_path)
-        publish_on_data_gouv(public_url, archive_size, archive_sha1)
-        return True
+        for area in areas_list:
+            print(f"Processing area: {area}")
+            create_csv(directory_name, area)
+            (archive_path, archive_size, archive_sha1) = create_archive(
+                directory_name, area
+            )
+            # Delete file after archiving
+            os.remove(f"{file_path(directory_name, area)}.csv")
```

Comment:
Plutôt que de mettre des `os.remove`, je commencerais la boucle for par `cleanup_directory(directory_name)` pour être sûr de partir d'un répertoire vide.

Reply:
Le problème est que le `cleanup_directory` supprime le contenu et le répertoire.

Reply:
fait dans https://github.com/fab-geocommuns/RNB-coeur/pull/393/commits/eaa93db9bb87620893da6e588a0101f25130f5ab